### PR TITLE
UINV-159 Display voucher export details accordion

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
https://issues.folio.org/browse/UINV-159

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/settings/BatchGroupConfigurationSettings/BatchVoucherExportsList.js [hunk 1]
```diff
@@ -27,9 +27,11 @@ export function BatchVoucherExportsList({ batchVoucherExports, format, onNeedMor
         <FolioFormattedTime dateString={d.end || d.start} />
       ),
       // eslint-disable-next-line react/prop-types
-      exportButton: ({ batchVoucherId }) => (
+      exportButton: ({ batchVoucherId, metadata }) => (
         <ExportVoucherButton
           batchVoucherId={batchVoucherId}
+           // eslint-disable-next-line react/prop-types
+          fileName={metadata?.createdDate || batchVoucherId}
```
Comment (aliaksei-chumakou) [inline src/settings/BatchGroupConfigurationSettings/BatchVoucherExportsList.js:34]: Are there any cases when `metadata` is undefined? IIUC, `batchVoucherId` is not desired filename at all

File: src/invoices/InvoiceDetails/InvoiceDetailsContainer.js [hunk 2]
```diff
@@ -62,12 +68,38 @@ function InvoiceDetailsContainer({
...
+          vendorResp,
+          invoiceLinesResp,
+          approvalsConfigResp,
+          batchVoucherExportResp,
+          exportConfigsResp,
+        ]) => {
           setVendor(vendorResp);
           setInvoiceLines(invoiceLinesResp);
+          setBatchVoucherExport(batchVoucherExportResp[0]);
+          setExportFormat(exportConfigsResp[0]?.format);
```
Comment (aliaksei-chumakou) [inline src/invoices/InvoiceDetails/InvoiceDetailsContainer.js:102]: IMHO, it's kind a wrong logic. if config has no format or it's absent we shouldn't display table at all

Comment (AndreiShumski2) [reply src/invoices/InvoiceDetails/InvoiceDetailsContainer.js:102]: `format` is required for exportConfig

File: src/invoices/InvoiceDetails/InvoiceBatchVoucherExport/InvoiceBatchVoucherExport.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      </Col>
+
+      <Col xs={4}>
+        <KeyValue
+          label={<FormattedMessage id="ui-invoice.invoice.details.batchVoucherExport.fileName" />}
+        >
+          <FolioFormattedTime dateString={batchVoucherExport.metadata.createdDate} />
+          <ExportVoucherButton
+            batchVoucherId={batchVoucherExport.batchVoucherId}
+            fileName={batchVoucherExport.metadata.createdDate}
```
Comment (aliaksei-chumakou) [inline src/invoices/InvoiceDetails/InvoiceBatchVoucherExport/InvoiceBatchVoucherExport.js:35]: Are you sure that back-end sends the right date format? I doubt it

Comment (AndreiShumski2) [reply src/invoices/InvoiceDetails/InvoiceBatchVoucherExport/InvoiceBatchVoucherExport.js:35]: BE sends data `2020-08-07T07:23:33.528+0000` that converts to `2020-08-07T07_23_33_528+0000`

File: src/settings/BatchGroupConfigurationSettings/BatchVoucherExportsList.js (hunk 1 above)
Comment (AndreiShumski2) [reply src/settings/BatchGroupConfigurationSettings/BatchVoucherExportsList.js:34]: I faced it in tests only

File: src/invoices/InvoiceDetails/InvoiceBatchVoucherExport/InvoiceBatchVoucherExport.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+} from '@folio/stripes/components';
+import {
+  FolioFormattedTime,
+} from '@folio/stripes-acq-components';
+
+import { BATCH_VOUCHER_EXPORT_STATUS_LABEL } from '../../../common/constants';
+import ExportVoucherButton from '../../../settings/BatchGroupConfigurationSettings/ExportVoucherButton';
+import BatchGroupValue from '../BatchGroupValue';
+
+const InvoiceBatchVoucherExport = ({ batchVoucherExport, exportFormat }) => {
```
Comment (NikitaSedyx) [inline src/invoices/InvoiceDetails/InvoiceBatchVoucherExport/InvoiceBatchVoucherExport.js:18]: just `batchVoucher` is better IMO

File: src/invoices/InvoiceDetails/InvoiceDetails.js [hunk 5]
```diff
@@ -222,6 +225,17 @@ function InvoiceDetails({
                 accountingCode={invoice.accountingCode}
               />
             </Accordion>
+            {showVoucherInformation && batchVoucherExport && (
```
Comment (NikitaSedyx) [inline src/invoices/InvoiceDetails/InvoiceDetails.js:228]: I think @aliaksei-chumakou  mostly related to this line, if there is no export config - no way to export

Comment (AndreiShumski2) [reply src/invoices/InvoiceDetails/InvoiceDetails.js:228]: In my understanding, no export config -> no batch voucher exports.

Comment (aliaksei-chumakou) [reply src/invoices/InvoiceDetails/InvoiceDetails.js:228]: but you can't tell that looking at your code

Comment (NikitaSedyx) [reply src/invoices/InvoiceDetails/InvoiceDetails.js:228]: are you sure in it that without export BE can't return batchVoucherExport info?

Comment (AndreiShumski2) [reply src/invoices/InvoiceDetails/InvoiceDetails.js:228]: batchVoucherExport is based on exportConfig

**Pull request merged**